index_trade.html

```diff
@@ -0,0 +1,263 @@
+    
+  // FETCH DATA
+  d3.json('data-exch.json', function (error, json) {
+    if (error) { throw error; }
+    data = json;
+    setup();
+    update();
+  });
+
+    
+  function setup() {
+    // SVG
+    svg = d3.select('#chart')
+      .append('svg')
+      .attr('width', width + margin.right + margin.left)
+      .attr('height', height + margin.top + margin.bottom)
+      .append('g')
+      .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
+      
+    // SCALES
+      var xmin = d3.min(data, function (d) { return d.trade; }) - 500;
+      
+      x = d3.scale.linear()
+        .domain([xmin, -xmin])
+        .range([0, width])
+        .nice();
+
+    y = d3.scale.linear()
+      .domain([0, d3.max(data, function (d) { return d.rate; })])    
+      .range([height, 0])    
+      .nice();
+      
+    // AXES
+      var xAxis = d3.svg.axis()
+      .scale(x)
+      .tickSize(-height);
+      
+      var yAxis = d3.svg.axis()
+      .scale(y)
+      .tickSize(-width)
+      .orient('left');   
+      
+      svg.append('g')
+          .attr('class', 'x axis')
+          .call(xAxis)
+          .attr('transform', 'translate(0,' + (height) + ')')
+        .append('text')
+          .attr('x', width/2)
+          .attr('y', 70)
+          .style('text-anchor', 'middle')
+          .text('Trade Balance Between US and UK');      
+      
+      svg.append('g')
+          .attr('class', 'y axis')
+          .call(yAxis)
+        .append('text')
+          .attr('transform', 'rotate(-90)')
+          .attr('y', -60)
+          .attr('x', -height/2)      
+          .style('text-anchor', 'middle')      
+          .text('Exchange Rate, GBP/USD');           
+      
+      // GRID LINES
+      
+      var ygrid = svg.append('g')
+      .attr('class', 'y axis')
+      .call(yAxis);
+      
+      ygrid.selectAll('g').filter(function(d) { return d; })
+        .classed('minor', true);
+       
+    // EVENT HANDLERS
+    d3.select('#year-range').on('change', function () {
+      options.yearRange = d3.event.target.value;
+      update();
+    });
+  }
+    
+  function update() {
+    txData = data.slice();
+    if (options.yearRange !== 'all') {
+      var yearRange = options.yearRange.split('-');
+      txData = txData.filter(function (d) {
+        return d.year >= yearRange[0] && d.year <= yearRange[1];
+      });
+    }
+      
+    // SCATTERPLOT
+    var points = svg.selectAll('.point')
+      .data(txData);
+    points.enter().append('circle')
+      .attr('class', 'point')
+      .attr('r', 5);
+    points
+      .attr('cx', function (d) { return x(d.trade); })
+      .attr('cy', function (d) { return y(d.rate); });
+    points.exit().remove();
+
+      // HISTOGRAM
+      
+      // x hist
+      
+    var xmap = data.map(function(d) { return d.trade; })
+    var xhist = d3.layout.histogram().bins(x.ticks())(xmap);      
+      
+      console.log('hist', xhist);
+      
+      bwidth = x(x.ticks()[1])/2;
```

Comment:
Of course, we could get this using `dx` from the histogram layout. But this works too.